src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java

```diff
@@ -340,13 +340,14 @@ public ExamResource(UserRepository userRepository, CourseRepository courseReposi
         }
 
         Set<Exercise> exercises = new HashSet<>();
+        List<TutorParticipation> tutorParticipations = new ArrayList<>();
         // extract all exercises for all the exam
         for (ExerciseGroup exerciseGroup : exam.getExerciseGroups()) {
             exerciseGroup.setExercises(courseRepository.getInterestingExercisesForAssessmentDashboards(exerciseGroup.getExercises()));
             exercises.addAll(exerciseGroup.getExercises());
+            tutorParticipations.addAll(tutorParticipationRepository.findAllByAssessedExerciseExerciseGroupIdAndTutorId(exerciseGroup.getId(), user.getId()));
```

Comment:
I would suggest to move this out of the for loop and use examId instead

Reply:
this way the performance will be better in case many exercise groups exist

Reply:
ahh, thank you! Since `@EntityGraph` is too abstract for me, I could not think of using examId would solve the case. I applied the suggestion.